deprecate using namespace std

molecula451 opened the pull request:
Related: https://github.com/ethereum/solidity/issues/14403

This PR deprecate `using namespace std` in some of the test folders
- [x]  test/contracts
- [x]  test/libevmasm
- [x]  test/liblangutil
- [x]  test/libsolutil
- [x] test/libsolidity

molecula451 committed:
deprecate using namespace std

test: updat filereader test

deprecate namespace std

deprecate namespace std

deprecate namespace std

deprecate namespace std

deprecate namespace std

deprecate namespace std

deprecate namespace std

deprecate namespace std

deprecated std namespace

deprecated std namespace

deprecated std namespace

deprecated std namespace

deprecated std namespace

deprecated std namespace

deprecated std namespace

deprecated std namespace

deprecated std namespace

depecrate namespace std

deprecated namespace std

check ci

clean line

Co-authored-by: Nikola Matić <nikola.matic@ethereum.org>

purge line

pure line

deprecate std

deprecate std

deprecate std

deprecate std

deprecate std

deprecate

deprecate std

bye namespace

molecula451 commented:
@nikola-matic @cameel

molecula451 commented:
i'm buildin on linux, prob a bug on the ps script? https://app.circleci.com/pipelines/github/ethereum/solidity/31091/workflows/809e77d3-ad23-499c-a75f-966336a9af60/jobs/1383651

nikola-matic commented on test/contracts/AuctionRegistrar.cpp:
```diff
@@ -32,7 +32,7 @@
 #include <optional>
 
-using namespace std;
+
```
```suggestion
```

Please go over all of the files and remove newly introduced newlines such as this.

nikola-matic commented on test/libsolutil/JSON.cpp:
```diff
@@ -26,7 +26,7 @@
 #include <boost/test/unit_test.hpp>
 
-using namespace std;
+
```
Yeah, as mentioned before - just delete the `using namespace std;` instance, without replacing it with a newline.

molecula451 replied:
no new lines :P

nikola-matic commented on test/libsolidity/interface/FileReader.cpp:
```diff
@@ -226,10 +226,10 @@ BOOST_AUTO_TEST_CASE(normalizeCLIPathForVFS_stripping_root_name)
-		for (auto convert: {boost::to_lower_copy<string>, boost::to_upper_copy<string>})
+		for (auto convert: {boost::to_lower_copy<std::string>, boost::to_upper_copy<std::string>})
 		{
 			boost::filesystem::path workDirWin = convert(root, locale()) / workDir.relative_path();
```
```suggestion
			boost::filesystem::path workDirWin = convert(root, std::locale()) / workDir.relative_path();
```

Windows build is failing due to this.

nikola-matic reviewed (commented):
I've taken a quick look - and aside from addressing the comments, the actual `std::` additions seem good - i.e. you've not replaced any comments/strings.

In addition, you'll have to add the directories you've changed to the [check_style.sh](https://github.com/ethereum/solidity/pull/14522/files#diff-e0362b21ff706f362633472037a5d020e22eccc7ca5595031d460e8b664c2ad8) script (I've linked a similar PR).

Also, please squash the commits into one - and if you're unsure how to do this, drop a message in this PR, and I'll do it. Thanks for helping out :)

molecula451 commented:
> I've taken a quick look - and aside from addressing the comments, the actual `std::` additions seem good - i.e. you've not replaced any comments/strings.
> 
> In addition, you'll have to add the directories you've changed to the [check_style.sh](https://github.com/ethereum/solidity/pull/14522/files#diff-e0362b21ff706f362633472037a5d020e22eccc7ca5595031d460e8b664c2ad8) script (I've linked a similar PR).
> 
> Also, please squash the commits into one - and if you're unsure how to do this, drop a message in this PR, and I'll do it. Thanks for helping out :)

done over here, you have the power to squash the last commit and merge :]
